packages/core/test/setup.types.test.ts

```diff
@@ -2184,9 +2184,48 @@ describe('setup()', () => {
 
     const actor = createActor(machine);
 
-    actor.getSnapshot().getMeta().a satisfies { layout: string } | undefined;
+    actor.getSnapshot().getMeta()['(machine)'] satisfies
+      | { layout: string }
+      | undefined;
 
     // @ts-expect-error
     actor.getSnapshot().getMeta().a?.whatever;
   });
+
+  it('should strongly type the state IDs in snapshot.getMeta()', () => {
```

Comment:
this PR introduces very similar tests in 2 places - are there any different? It seems like keeping this in a single place would be better

Reply:
Consolidated here: [e79ad4e](https://github.com/statelyai/xstate/pull/4979/commits/e79ad4e223053310e9d63e9192198cf19df7b979)